mapie/control_risk/p_values.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+from numpy.typing import NDArray
+from typing import Iterable, Union
+from scipy.stats import binom
+
+
+def hoefdding_bentkus_p_value(
+    r_hat: NDArray,
+    n: int,
+    alpha: Union[float, NDArray]
+) -> NDArray:
+    """
+    Parameters
+    ----------
+    r_hat : NDArray of shape (len(lambdas), )
+        Empirical risk of metric_control with respect
+        to the lambdas.
+    n : Integer value
+        Correspond to the number of observations in
+        X_cal dataset.
+    alpha: NDArray.
+        Correspond to the value that r_hat should not
+        exceed.
+
+    Method
+    -------
+    The method computes the p_values according to
+    the Hoeffding_Bentkus inequality for each
+    alpha.
+    We return the min between the Hoeffding and
+    Bentkus p-values (Note that it depends on
+    scipy.stats).
+    Returns
+    -------
+    p_values: NDArray of shape
+        (len(lambdas), len(alpha)).
+    """
+    # bentkus_p_value = np.e * binom.cdf(np.ceil(n*r_hat), n, alpha)
```

Comment:
STYLE: if ignored, it should be deleted.

Reply:
Done